platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -158,34 +64,36 @@ export function ViewportGridProvider({ children, service }) {
 
           const displaySetOptions = updatedViewport.displaySetOptions || [];
           if (!displaySetOptions.length) {
-            // Copy all the display set options, assuming a full set of displa
-            // set UID's is provided.
+            // Copy all the display set options, assuming a full set of displayset UID's is provided.
             displaySetOptions.push(...previousViewport.displaySetOptions);
             if (!displaySetOptions.length) {
               displaySetOptions.push({});
             }
           }
 
-          let newViewport = {
+          const newViewport = {
             ...previousViewport,
             displaySetInstanceUIDs,
             viewportOptions,
```

Comment:
Should the viewportOptions and displaySetOptions be a deep copy here - or at least a copy of those values, since you are modifying the viewportOptions down below.